### DISCUSSION TITLE
Support for thousands of projects

### DISCUSSION THREAD

**Discussion by erikvullings** [asreview/asreview#782] [Q&A]
I have the following use case, and was wondering if that is supported by ASReview: at our company (TNO), we have a few thousand projects. For each of these projects, I would like to automatically create an ASReview project, using the same database (consisting of internal publications and external publications). So my questions are:
- Does ASReview support thousands of projects, or is there a limit. 
- Does it become very slow when there are thousands of projects.
- Since I want to use the same database for every TNO project, will that database file be copied every time? If so, should I create a plugin, like the COVID plugin, or can I use symbolic linking?
- How can I seamlessly update the database for each project on a regular basis.

**Comment by Rensvandeschoot** [comment]
Dear @erikvullings ,

Thank you for reaching out to us and great to read you are considering to use ASReview on such a large scale!! So far, we have not tested with so many projects, but let's see if together we can figure out if it will work!

1. @J535D165 (Jonathan is our lead engineer) Do you know if there is a technical limitation to the number of project files?

2. @terrymyc (Terry is our front-end developer) Are there limitations in the front-end? I can imagine the new project dashboard (to be released in version 1 - expected November 2021) is much better equipped to deal with so many projects, right? @erikvullings, see for screenshots of the new project dashboard this [PR](https://github.com/asreview/asreview/pull/624). The two of you might want to chat to see if there are features essential to handle so many projects.

3. If you have many projects the underlying log-file to completely reproduce the proces might become quite large, so you might run into storage issues. @PeterLombaers (Peter is one of our back-end developers) has solved this by changing the way data is stored (labelling decisions, model settings and ranking at each iteration of the model), see this [PR](https://github.com/asreview/asreview/pull/694). Again, you would need version 1 to make use of it. 


To be continued... 

Best
Rens
